### PR TITLE
Flyway Implementation

### PR DESCRIPTION
**Pull request by gdesiato**
- addition of flyway maven dependency
- creation of V1__initial_schema.sql file
- application.properties modified for implementing flyway

### CODE REVIEW

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+--
+
+DROP TABLE IF EXISTS `user`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `user` (
+  `id` bigint NOT NULL AUTO_INCREMENT,
+  `email` varchar(255) NOT NULL,
+  `password` varchar(255) NOT NULL,
+  `username` varchar(255) NOT NULL,
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:108]: Do you still need this?

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `adsr` (
+  `id` bigint NOT NULL AUTO_INCREMENT,
+  `attack_time` double NOT NULL,
+  `decay_time` double NOT NULL,
+  `release_time` double NOT NULL,
+  `sustain_level` double NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=MyISAM DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:32]: Why not default engine InnoDB?

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 3]
```diff
@@ -0,0 +1,151 @@
+-- MySQL dump 10.13  Distrib 8.0.30, for macos12 (x86_64)
+--
+-- Host: localhost    Database: puresynth
+-- ------------------------------------------------------
+-- Server version	8.0.30
+
+/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:7]: What are these: `40101 ...`?

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+/*!50503 SET NAMES utf8mb4 */;
+/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
+/*!40103 SET TIME_ZONE='+00:00' */;
+/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
+/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
+/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
+/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
+
+--
+-- Table structure for table `adsr`
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:19]: Which tables do you really need and use currently in the app?

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  `waveform_type` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=MyISAM DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `roles`
+--
+
+DROP TABLE IF EXISTS `roles`;
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:87]: Weren't we supposed to drop those?

File: src/main/resources/db.migration/V1__initial_schema.sql [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
+/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
+/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
+/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
+/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
+/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
+/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;
+
+-- Dump completed on 2024-03-24 11:02:40
```
Comment (j-nowak) [inline src/main/resources/db.migration/V1__initial_schema.sql:151]: You should clean this file and don't use plain dump.

Comment (gdesiato) [reply src/main/resources/db.migration/V1__initial_schema.sql:151]: ok, I'll clean the file and edit it.

File: pom.xml [hunk 7]
```diff
@@ -61,6 +61,10 @@
 			<version>5.9.2</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.flywaydb</groupId>
+			<artifactId>flyway-mysql</artifactId>
+		</dependency>
```
Comment (j-nowak) [inline pom.xml:67]: Let's not do mixed things in the PR. Big PRs are one of the main reasons, why software fails on production. I know we don't have this risk, but I want to teach you good practices.

PRs should be small and self-contained. They should have focused, singular goals:
- if there is a clean-up, let's not add new features
- if we're adding new code, let's not do major refactoring.
I recommend you look for guides and articles on how to structure and split PRs into small scopes. Another useful tactic it is to stack PRs on top of each other (useful while waiting for a review).

In this situation let's split into 2 PRs:
1. Clean-up, removing old code and controllers.
2. Adding Flyway integration.

File: src/main/java/com/desiato/puresynth/configurations/SecurityConfig.java [hunk 8]
```diff
@@ -9,37 +9,37 @@
...
-        http.csrf(
-                c -> c.disable()
-        );
-
-        return http.build();
-    }
-
-}
+//@Configuration
+//@EnableWebSecurity(debug = true)
```
Comment (j-nowak) [inline src/main/java/com/desiato/puresynth/configurations/SecurityConfig.java:13]: Why is it commented out?

File: pom.xml (hunk 7 above)
Comment (gdesiato) [reply pom.xml:67]: ok, Thanks! That makes totally sense

File: src/main/java/com/desiato/puresynth/configurations/SecurityConfig.java (hunk 8 above)
Comment (gdesiato) [reply src/main/java/com/desiato/puresynth/configurations/SecurityConfig.java:13]: I commented that out momentarily to check the app, before cleaning it up as well.
I need to modify it, after removing roles and admin users

**Pull request closed without merging**